### ISSUE TITLE
Tracking WIP for Spaces

### ISSUE THREAD

**Issue by GeorgeS2019** [ChengYen-Tang/RL.Env#1]
## [Spaces](https://github.com/ChengYen-Tang/RL.Env/tree/master/RL.Env/Spaces)
- [X] box
- [ ] dict
- [X] discrete
- [ ] graph
- [X] multi_binary
- [X] multi_discrete
- [ ] sequence
- [X] space
- [ ] text
- [ ] tuple
- [ ] utils

**Comment by GeorgeS2019** [comment]
@ChengYen-Tang 

I understand your goal is [sb3](https://github.com/DLR-RM/stable-baselines3/tree/master/stable_baselines3) for .net

I hope u could learn from TorchSharp, there are right now so many reinforcement learning projects for .net, many are classical BUT so far I have not convinced anyone of they could become e.g. the "standard" for .NET the way OpenAI's Gym and SB3 have done for the python communities. 

Why I believe why you are attempting is GREAT!

**Comment by GeorgeS2019** [comment]
If you could adhere to e.g. file structures of OpenAI Gym and SB3 (the way we try to keep TorchSharp adhered to PyTorch), this would really help to draw more confidence in your attempt!!

**Comment by GeorgeS2019** [comment]
Please use TorchSharp to promote your work and draw more people to your attempt. 

If possible, use Godot (no 1 MIT FOSS) as the 3D engine instead of Unity.

**Comment by ChengYen-Tang** [comment]
> If you could adhere to e.g. file structures of OpenAI Gym and SB3 (the way we try to keep TorchSharp adhered to PyTorch), this would really help to draw more confidence in your attempt!!

I disagree because these two projects do not follow the rules of OOP, which is why I keep hitting bottlenecks.
And their code maintainability is very low, especially SB3. At this stage, I hope to replicate the functions inside as much as possible. In the next stage, I think it is necessary to refactor and improve performance.

> Please use TorchSharp to promote your work and draw more people to your attempt.
> 
> If possible, use Godot (no 1 MIT FOSS) as the 3D engine instead of Unity.

I have no plans to port the game at the moment. The algorithm of reinforcement learning is already a huge project.😂

**Comment by GeorgeS2019** [comment]
> I have no plans to port the game at the moment. The algorithm of reinforcement learning is already a huge project.😂

Once u have a simple working classical tutorial or sample used in Gym, I see what I can do to port it to Godot.

An example of [SB3 using Godot using c# and exported onnx](https://github.com/JimFFM/ml-hackathon-2021/blob/main/README.md)
![uzYkLCgQH1](https://user-images.githubusercontent.com/59052120/142358633-33c33c02-7e19-4ce0-b9a8-1148e8c8017d.gif)

**Comment by GeorgeS2019** [comment]
@ChengYen-Tang 
Did u remove RL.Algothrim?

**Comment by ChengYen-Tang** [comment]
I make it a private project
